src/target/riscv/field_helpers.h

```diff
@@ -0,0 +1,34 @@
+/* SPDX-License-Identifier: GPL-2.0-or-later */
+
+#ifndef FIELD_HELPERS_H
+#define FIELD_HELPERS_H
+
+#include <stdint.h>
+#include <assert.h>
+
+static inline uint64_t get_field(uint64_t reg, uint64_t mask)
+{
+	return (reg & mask) / (mask & ~(mask << 1));
+}
+
+static inline uint32_t get_field32(uint64_t reg, uint64_t mask)
+{
+	uint64_t value = get_field(reg, mask);
+	assert(value <= UINT32_MAX);
+	return value;
+}
+static inline uint64_t set_field(uint64_t reg, uint64_t mask, uint64_t val)
```

Comment:
> Tim: [...] The only place where the return type matters is in printf-style functions, and I made get_value32() for those cases where a change was needed.

Actually, compilers or static analysis tools may complain about "implicit narrowing" when the 64-bit return value is assigned to a 32-bit value. 

For example clang-tidy has rules for it: https://clang.llvm.org/extra/clang-tidy/checks/cppcoreguidelines/narrowing-conversions.html

GCC has the`-Wconversion` option for it:
```
main.c: In function ‘main’:
main.c:6:24: warning: conversion from ‘uint64_t’ {aka ‘long unsigned int’} to ‘uint32_t’ {aka ‘unsigned int’} may change value [-Wconversion]
  uint32_t second_val = val;
```

For that reason I would recommend to create also the `set_field32()` variant. To have it ready for use in case any of those warnings pop up.

Reply:
I just tried a build with clang and I got no warnings. But I've added a helper for it.

Reply:
Thanks.